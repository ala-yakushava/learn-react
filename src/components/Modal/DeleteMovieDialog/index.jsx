import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';

import './style.scss';
import Heading from '../../Heading';
import ModalFooter from '../ModalFooter';
import Button from '../../Button';
import { removeMovie } from '../../../slices/moviesInfo';

const DeleteMovieDialog = ({ movieId }) => {
  const dispatch = useDispatch();

  const handleClick = () => {
    dispatch(removeMovie(movieId));
  };

  return (
    <div className="DeleteMovieDialog">
      <Heading>Delete Movie</Heading>
      <p>Are you sure you want to delete this movie?</p>
      <ModalFooter>
        <Button mode="primary" onClick={handleClick}>Confirm</Button>
      </ModalFooter>
    </div>
  );
};

DeleteMovieDialog.propTypes = {
  movieId: PropTypes.number.isRequired,
};

export default DeleteMovieDialog;
