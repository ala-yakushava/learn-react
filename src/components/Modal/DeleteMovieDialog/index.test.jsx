import React from 'react';
import renderer from 'react-test-renderer';

import DeleteMovieDialog from './index';

describe('Components - DeleteMovieDialog', () => {
  const handleClick = jest.fn();

  test('should match snapshot', () => {
    const tree = renderer
      .create(
        <DeleteMovieDialog onClick={handleClick} />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
