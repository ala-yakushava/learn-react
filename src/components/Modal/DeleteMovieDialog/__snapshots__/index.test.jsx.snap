// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - DeleteMovieDialog should match snapshot 1`] = `
<div
  className="DeleteMovieDialog"
  data-testid="delete-dialog"
>
  <div
    className="Heading"
  >
    Delete Movie
  </div>
  <p>
    Are you sure you want to delete this movie?
  </p>
  <div
    className="ModalFooter"
  >
    <button
      className="Button Button--primary"
      disabled={false}
      onClick={[MockFunction]}
      type="button"
    >
      <span
        className="Button_text"
      >
        Confirm
      </span>
    </button>
  </div>
</div>
`;
